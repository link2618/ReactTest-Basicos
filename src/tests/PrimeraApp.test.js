import React from 'react'
import '@testing-library/jest-dom'
// import { render } from '@testing-library/react'
import PrimeraApp from '../PrimeraApp'
import { shallow } from 'enzyme'

describe('Pruebas de Primera App', () => {
    // test('Debe mostrar el mensaje "Hola mundo"', () => {
        // const saludo = "Hola mundo"
        // const wrapper = render(<PrimeraApp saludo={saludo}/>)

        // expect(wrapper.getByText(saludo)).toBeInTheDocument()
    // })
    test('Debe mostrar el mensaje "Hola mundo"', () => {
        const saludo = "Hola mundo"
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot()
    })

    test('Debe mostrar el subtitulo cuando se envia', () => {
        const saludo = "Hola mundo"
        const subtitulo = 'Soy el subtitulo enviado'

        const wrapper = shallow(
        <PrimeraApp saludo={saludo} subtitulo={subtitulo}/>
        )

        // Se pone p por que estamos buscando la etiqueta p
        const textoParrafo = wrapper.find('p').text()

        expect(textoParrafo).toBe(subtitulo)
    })
        
})
