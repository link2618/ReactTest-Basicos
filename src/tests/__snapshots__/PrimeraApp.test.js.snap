// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Primera App Debe mostrar el mensaje "Hola mundo" 1`] = `
<Fragment>
  <h1>
    Hola mundo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas de Primera App Debe mostrar el subtitulo cuando se envia 1`] = `
<Fragment>
  <h1>
    Hola mundo
  </h1>
  <p>
    Soy el subtitulo eanviado
  </p>
</Fragment>
`;
